File: la_descente/ladescente.swift
import Glibc
import Foundation

public struct StderrOutputStream: TextOutputStream {
    public mutating func write(_ string: String) { fputs(string, stderr) }
}
public var errStream = StderrOutputStream()

/**
 * The while loop represents the game.
 * Each iteration represents a turn of the game
 * where you are given inputs (the heights of the mountains)
 * and where you have to print an output (the index of the mountain to fire on)
 * The inputs you are given are automatically updated according to your last actions.
 **/


// game loop
while true {
    var max: Int = 0
    var index: Int = 0
    for i in 0...7 {
        let mountainH = Int(readLine()!)! // represents the height of one mountain.
        if (max < mountainH) {
            max = mountainH
            index = i
        }
    }

    // Write an action using print("message...")
    // To debug: print("Debug messages...", to: &errStream)

    print(index)     // The index of the mountain to fire on.

}

File: la_descente/ladescente.js
/**
 * The while loop represents the game.
 * Each iteration represents a turn of the game
 * where you are given inputs (the heights of the mountains)
 * and where you have to print an output (the index of the mountain to fire on)
 * The inputs you are given are automatically updated according to your last actions.
 **/

// game loop
while (true) {
    const mountains = [];
    for (let i = 0; i < 8; i++) {
        const mountainH = parseInt(readline()); // represents the height of one mountain.
        mountains.push(mountainH);
    }

    // Write an action using console.log()
    // To debug: console.error('Debug messages...');
    let max = 0;
    let index = 0;
    for (let i = index; i < 8; i++) {
        if (max < mountains[i]) {
            max = mountains[i];
            index = i;
        }
    }
    console.log(index);     // The index of the mountain to fire on.
}

